core/store/src/flat_state.rs

```diff
@@ -792,16 +781,30 @@ impl FlatStorageStateInner {
         FlatStorageError::BlockNotSupported((self.flat_head, *block_hash))
     }
 
-    /// Get deltas between blocks `target_block_hash`(inclusive) to flat head(exclusive),
+    /// Gets delta for the given block and shard `self.shard_id`.
+    fn get_delta(&self, block_hash: &CryptoHash) -> Result<Arc<FlatStateDelta>, FlatStorageError> {
+        // TODO (#7327): add limitation on cached deltas number to limit RAM usage
+        // and read single `ValueRef` from delta if it is not cached.
+        Ok(self
+            .deltas
+            .get(block_hash)
+            .ok_or(self.create_block_not_supported_error(block_hash))?
+            .clone())
+    }
+
+    /// Get sequence of blocks `target_block_hash` (inclusive) to flat head (exclusive)
     /// in backwards chain order. Returns an error if there is no path between them.
-    fn get_deltas_between_blocks(
+    fn get_blocks_to_head(
         &self,
         target_block_hash: &CryptoHash,
-    ) -> Result<Vec<Arc<FlatStateDelta>>, FlatStorageError> {
-        let flat_head_info = self.blocks.get(&self.flat_head).unwrap();
+    ) -> Result<Vec<CryptoHash>, FlatStorageError> {
+        let flat_head_info = self
+            .blocks
+            .get(&self.flat_head)
+            .ok_or(self.create_block_not_supported_error(&self.flat_head))?;
```

Comment:
I think here we should unwrap instead of returning the block not supported error since this means flat storage is in an inconsistent state. Flat head should always be part of self.blocks.